config.rb

```diff
@@ -25,7 +25,26 @@
   blog.per_page = 10
 end
 
-page "/blog/feed.xml", layout: false
+# middleman-blog does not allow the `mount_at_root` like feature
+# so we define two blog instances for English one and localized one.
+
+# English blog at /blog
+activate :blog do |blog|
+  blog.name = 'en'
+  # This will add a prefix to all links, template references and source paths
+  blog.prefix = "blog"
+  configure_blog(blog)
+end
+
+# Localized blogs at /{lang}/blog
+activate :blog do |blog|
+  blog.name = 'i18n'
```

Comment:
> This does that anyway, as it still takes en into account.

It doesn't generate `/en` namespace because `mount_at_root: :en` option is specified (for the `/blog` index page) and the `current_blog_name` helper returns the root blog name `'en'` for `en` locale (for each permalink).

```bash
$ bundle exec middleman build
# ...
$ ls -l build
total 488
drwxr-xr-x+  3 nkymyj  staff    102 Nov 13 10:47 404/
drwxr-xr-x+  3 nkymyj  staff    102 Nov 13 10:47 about/
-rw-r--r--+  1 nkymyj  staff   8530 Nov 13 10:47 apple-touch-icon-114x114-precomposed.png
-rw-r--r--+  1 nkymyj  staff   3389 Nov 13 10:47 apple-touch-icon-120x120-precomposed.png
-rw-r--r--+  1 nkymyj  staff  12093 Nov 13 10:47 apple-touch-icon-144x144-precomposed.png
-rw-r--r--+  1 nkymyj  staff  13279 Nov 13 10:47 apple-touch-icon-152x152-precomposed.png
-rw-r--r--+  1 nkymyj  staff   3677 Nov 13 10:47 apple-touch-icon-57x57-precomposed.png
-rw-r--r--+  1 nkymyj  staff   3905 Nov 13 10:47 apple-touch-icon-60x60-precomposed.png
-rw-r--r--+  1 nkymyj  staff   4886 Nov 13 10:47 apple-touch-icon-72x72-precomposed.png
-rw-r--r--+  1 nkymyj  staff   5250 Nov 13 10:47 apple-touch-icon-76x76-precomposed.png
-rw-r--r--+  1 nkymyj  staff   3677 Nov 13 10:47 apple-touch-icon-precomposed.png
-rw-r--r--+  1 nkymyj  staff   3677 Nov 13 10:47 apple-touch-icon.png
drwxr-xr-x+  9 nkymyj  staff    306 Nov 13 10:47 blog/
drwxr-xr-x+  3 nkymyj  staff    102 Nov 13 10:47 contributing/
drwxr-xr-x+ 10 nkymyj  staff    340 Nov 13 10:47 documentation/
-rw-r--r--+  1 nkymyj  staff  14071 Nov 13 10:47 favicon-160x160.png
-rw-r--r--+  1 nkymyj  staff   1198 Nov 13 10:47 favicon-16x16.png
-rw-r--r--+  1 nkymyj  staff  19660 Nov 13 10:47 favicon-196x196.png
-rw-r--r--+  1 nkymyj  staff   1828 Nov 13 10:47 favicon-32x32.png
-rw-r--r--+  1 nkymyj  staff   6959 Nov 13 10:47 favicon-96x96.png
-rw-r--r--+  1 nkymyj  staff  24838 Nov 13 10:47 favicon.ico
-rw-r--r--+  1 nkymyj  staff   1198 Nov 13 10:47 favicon.png
drwxr-xr-x+  3 nkymyj  staff    102 Nov 13 10:47 help/
drwxr-xr-x+ 30 nkymyj  staff   1020 Nov 13 10:47 images/
-rw-r--r--+  1 nkymyj  staff   4462 Nov 13 10:47 index.html
drwxr-xr-x+  3 nkymyj  staff    102 Nov 13 10:47 ja/
drwxr-xr-x+  4 nkymyj  staff    136 Nov 13 10:47 javascripts/
-rw-r--r--+  1 nkymyj  staff  12093 Nov 13 10:47 mstile-144x144.png
-rw-r--r--+  1 nkymyj  staff  12874 Nov 13 10:47 mstile-150x150.png
-rw-r--r--+  1 nkymyj  staff  38957 Nov 13 10:47 mstile-310x310.png
-rw-r--r--+  1 nkymyj  staff   4678 Nov 13 10:47 mstile-70x70.png
drwxr-xr-x+  6 nkymyj  staff    204 Nov 13 10:47 stylesheets/
drwxr-xr-x+  3 nkymyj  staff    102 Nov 13 10:47 upgrading-from-rspec-2/
```